File: manashelper-bot/src/main/java/kg/manasuniversity/usbtypec/manashelper/entity/DailyMenu.java
package kg.manasuniversity.usbtypec.manashelper.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.LinkedHashSet;
import java.util.Set;
import java.util.UUID;

@Entity
@Table(name = "daily_menus")
@Getter
@Setter
@NoArgsConstructor
public class DailyMenu {
    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.UUID)
    private UUID id;

    @Column(name = "date", nullable = false)
    private LocalDate date;

    @Column(name = "views_count", nullable = false)
    private Integer viewsCount;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "daily_menu_dishes",
        joinColumns = @JoinColumn(name = "menu_id", nullable = false),
        inverseJoinColumns = @JoinColumn(name = "dish_id", nullable = false),
        uniqueConstraints = {
            @UniqueConstraint(name = "uk_daily_menu_dishes_menu_dish", columnNames = {"menu_id", "dish_id"})
        }
    )
    private Set<Dish> dishes;

    public DailyMenu(LocalDate date) {
        this.date = date;
        dishes = new LinkedHashSet<>();
        viewsCount = 0;
    }

    public void addDish(Dish dish) {
        dishes.add(dish);
    }

    public void clearDishes() {
        dishes.clear();
    }
}

File: src/manashelper/db/models/daily_menu.py
import datetime
import uuid
from typing import TYPE_CHECKING

from sqlalchemy import Column, ForeignKey, Table
from sqlalchemy.orm import Mapped, mapped_column, relationship

from manashelper.db.base import Base

if TYPE_CHECKING:
    from manashelper.db.models.dish import Dish

daily_menu_dishes = Table(
    "daily_menu_dishes",
    Base.metadata,
    Column("menu_id", ForeignKey("daily_menus.id"), primary_key=True),
    Column("dish_id", ForeignKey("dishes.id"), primary_key=True),
)


class DailyMenu(Base):
    __tablename__ = "daily_menus"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True)
    date: Mapped[datetime.date] = mapped_column(unique=True)
    views_count: Mapped[int] = mapped_column(default=0)

    dishes: Mapped[list["Dish"]] = relationship(secondary=daily_menu_dishes)
